src/js/hooks/productSupplier/form/useDeletePreferenceType.js

```diff
@@ -0,0 +1,97 @@
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+
+import { hideSpinner, showSpinner } from 'actions';
+import productSupplierApi from 'api/services/ProductSupplierApi';
+import confirmationModal from 'utils/confirmationModalUtils';
+
+const useDeletePreferenceType = ({
+  preferenceTypeData,
+  afterDelete,
+  isDefaultPreferenceType,
+  onCancel,
+}) => {
+  const dispatch = useDispatch();
+
+  const isPreferenceTypeEmpty = !_.some(
+    Object.values(_.omit(preferenceTypeData || {}, 'id')),
+  );
+
+  const modalLabels = {
+    title: {
+      label: 'react.productSupplier.deleteConfirmation.title.label',
+      default: 'Are you sure?',
+    },
+    content: isDefaultPreferenceType
+      ? {
+        label: 'react.productSupplier.defaultPreferenceDeleteConfirmation.label',
+        default: 'Are you sure you want to delete the default preference for this source?',
+      }
+      : {
+        label: 'react.productSupplier.preferenceTypeVariationsDeleteConfirmation.label',
+        default: 'Are you sure you want to delete this preference variation?',
+      },
+  };
+
+  const deletePreferenceType = async (onClose) => {
+    dispatch(showSpinner());
+    try {
+      if (preferenceTypeData?.id) {
+        await productSupplierApi.deleteProductSupplierPreference(preferenceTypeData?.id);
+      }
+      afterDelete?.();
```

Comment:
if we eventually don't call the delete (because of the if above), should we call the `afterDelete` anyway, or should it be inside the if also?

Reply:
yes, because if the row doesn't have the ID, it means it is a new row, so we don't have to make an API call, but we still would like to clear \ remove this row.

Reply:
thanks, this is what I wanted to hear